src/config.ts

```diff
@@ -139,3 +140,49 @@ export function getConfigParseResult(
 
   return configParseResult;
 }
+
+function getSuffixAppendingParseConfigHost(
```

Comment:
This is the heart of my changes. It solves the `appendSuffix` problem by providing TypeScript an interface to the file system that actually acts on the file names that ts-loader tracks, i.e. renamed with the `.ts` or `.tsx` suffix appropriately. The result is that `configParseResult.fileNames` _will_ include things like `foo.vue.tsx` when ts-loader is told to assume `foo.vue` is a TSX file.

> So I think this is actually a case where we can reliably be certain that the files that exist will _not_ be in the `tsconfig.json` and in fact cannot be.

_Au contraire mon frère_, @johnnyreilly 😄

Reply:
> This is the heart of my changes. It solves the appendSuffix problem by providing TypeScript an interface to the file system that actually acts on the file names that ts-loader tracks

That's clever! But could this be a bit unintuitive for the users?  Some patterns would match actual files but not transformed files. For instance, patterns like `src/**/*.vue`,  or just `main.vue` inside "files".  Right?

Of course the user can specify patterns on the transformed files. But then the `appendSuffix` is not that transparent anymore. Maybe that's not wrong, but still pretty confusing.  🤔